### PR TITLE
Added Firewall Saved Configuration functionality

### PR DESCRIPTION
**Pull request by dcoghlan**
Fixes issue #181

### CODE REVIEW

File: module/PowerNSX.psm1 [hunk 1]
```diff
@@ -27544,6 +27587,16 @@ function Get-NsxFirewallSavedConfiguration {
     Retrieves all saved Distributed Firewall configurations
 
     .EXAMPLE
+    Get-NsxFirewallSavedConfiguration TestBackup
+
+    Retrieves all Distributed Firewall configurations with the Name
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: typo ? (there is no name argument)

File: module/PowerNSX.psm1 [hunk 2]
```diff
@@ -27602,6 +27662,266 @@ function Get-NsxFirewallSavedConfiguration {
...
+    Update a saved Distributed Firewall configuration.
+
+    A copy of every published configuration is automatically saved as a draft. A
+    maximum of 100 configurations can be saved at a time. 90 out of
+    these 100 can be auto saved configurations from a publish operation.
+    When the limit is reached,the oldest configuration that is not marked for
+    preserve is purged to make way for a new one.
+
+    .EXAMPLE
+    Get-NsxFirewallSavedConfiguration -Name Change123 | Get-NsxFirewallSavedConfiguration -Preserve
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: Typo ... | Set-NsxFire... ?

File: module/PowerNSX.psm1 [hunk 3]
```diff
@@ -27602,6 +27662,266 @@ function Get-NsxFirewallSavedConfiguration {
...
+
+    Set the Preserve flag to true on an existing saved Distributed Firewall Configuration
+
+
+    .EXAMPLE
+    Get-NsxFirewallSavedConfiguration -Name Change123 | Get-NsxFirewallSavedConfiguration -Name Change123NewName -Description "Now With Description"
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: same issue

File: module/PowerNSX.psm1 [hunk 4]
```diff
@@ -27602,6 +27662,266 @@ function Get-NsxFirewallSavedConfiguration {
...
+
+    Update the name and description on an existing saved Distributed Firewall Configuration
+
+
+    .EXAMPLE
+    Get-NsxFirewallSavedConfiguration -Name Change123 | Get-NsxFirewallSavedConfiguration -Preserve
```
Comment (alagoutte) [inline module/PowerNSX.psm1]: same also

**Comment by dcoghlan** [comment]
jenkins test this please

**Pull request merged**